### PR TITLE
Hud refactor

### PR DESCRIPTION
**Pull request by Jezithyr**
Bugs TODO:
- [x] switching hands doesn't update item status panel (with x and by clicking) [space-wizards/space-station-14@`1438c6c` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/1438c6cda778ff764ced8c0db25a8bff69b13f49)
- [x] extra hands expand downwards outside the screen after the first row
- [x] actions given by items have incorrect icons [space-wizards/space-station-14@`abd31c4` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/abd31c4fc4d09c18d9c66cada1743e2ea49bac56)
- [x] Visual indicator for item use delay (i.e. the cooldown on a bikehorn) does not appear. [`cc98949` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/cc98949088ef53e59b4452a313a5956277e5d399)
- [x] switching between ghost, and human isn't clearing the bottom row correctly (just keeps adding more) [`677afb0` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/677afb0337f848a3ae42c2a230d3e14a6887f2f3)
- [x] possessing an npc as a ghost doesn't remove the ghost hud [`84c6389` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/84c638979e792789c3675990e70c0b201c2c690d)
- [x] toggling actions isn't showing state (ex: disarm doesn't change color)
- [x] ahelp UI (wrexbe: probably works now, but I'm not done yet https://github.com/space-wizards/space-station-14/pull/7202/commits/1dc2c2bd1c65f6688e1708e83e72c2c885464ff6)
- [x] inventory needs to be a second hotbar above the first one (wrexbe: [I asked around on discord, didn't find support for window inventory](https://discord.com/channels/310555209753690112/900426319433728030/1020028830343249922))
- [x] action hotkeys are offset by one [`83136c8` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/83136c8ee6c3200fe4762b1b38a199c7f874140d)
- [x] hotbar hotkeys are offset by one [`83136c8` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/83136c8ee6c3200fe4762b1b38a199c7f874140d)

### CODE REVIEW

**Comment by DrSmugleaf** [comment]
Closes https://github.com/space-wizards/space-station-14/issues/2531

**Comment by DrSmugleaf** [comment]
Closes https://github.com/space-wizards/space-station-14/issues/6865 (except the oldchat part)

**Review by moonheart08 — CHANGES_REQUESTED** [review]
You can ignore the doc comment ones i realized that's futile.

**Comment by DrSmugleaf** [comment]
> You can ignore the doc comment ones i realized that's futile.

Most of the functionality you pointed out to document this already existed before. This is a hud refactor, we are not going to document features we did not add.

File: Content.Client/Administration/UI/AdminMenuWindow.xaml.cs
Comment (DrSmugleaf) [reply Content.Client/Administration/UI/AdminMenuWindow.xaml.cs]: Done in [`4dfbf58` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/4dfbf58c0ac27e8335c7e88b1ca27392f5dc36d2)

File: Content.Client/CombatMode/CombatModeComponent.cs
Comment (DrSmugleaf) [reply Content.Client/CombatMode/CombatModeComponent.cs]: Done in [`4dfbf58` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/4dfbf58c0ac27e8335c7e88b1ca27392f5dc36d2)

File: Content.Client/CombatMode/CombatModeComponent.cs
Comment (DrSmugleaf) [reply Content.Client/CombatMode/CombatModeComponent.cs]: Done in [`4dfbf58` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/4dfbf58c0ac27e8335c7e88b1ca27392f5dc36d2)

File: Content.Client/CombatMode/CombatModeComponent.cs
Comment (DrSmugleaf) [reply Content.Client/CombatMode/CombatModeComponent.cs]: Done in [`4dfbf58` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/4dfbf58c0ac27e8335c7e88b1ca27392f5dc36d2)

File: Content.Client/CombatMode/CombatModeComponent.cs
Comment (DrSmugleaf) [reply Content.Client/CombatMode/CombatModeComponent.cs]: Done in [`4dfbf58` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/4dfbf58c0ac27e8335c7e88b1ca27392f5dc36d2)

**Review by ElectroJr — COMMENTED** [review]
All the `// TODO hud refactor` stuff obviously should get fixed before merging. For the minor TODOS, removing the comment opening a github issues might be fine.

AFAICT the chat box/window can't be resized anymore?

I know the action UI is hot garbage, but this PR still has some major changes to the action systems and it should definitely have been split into separate PRs whereever possible. On that note, some action issues that might hold up the PR:
- action icons no longer updated when toggled
- Picking up action granting items no longer auto populates the action bar.
- mapping actions & action loading from yaml are gone 
  - These are a pretty shitty makeshift tool for mappers and were always meant to be temporary. I don't actually know if they get used all that much. If the plan is to replace them, its probably fine to just drop them. But if they are used by mappers, and there is no replacement in sight, the functionality should stay.
- Action state handling changes no longer supports client-side actions.
  - Client exclusive actions were largely just added for mapping actions, so if they get removed & nothing else is likely to use them, you could also just remove the functionality.

File: Content.Client/Actions/ActionsSystem.cs
Comment (ElectroJr) [inline Content.Client/Actions/ActionsSystem.cs:54]: This should clone the server action, not add it directly. Otherwise, if the client modifies the action during prediction its actually modifying the server state, which will mess up prediction resetting.

File: Content.Shared/Inventory/InventoryTemplatePrototype.cs
Comment (ElectroJr) [inline Content.Shared/Inventory/InventoryTemplatePrototype.cs:22]: should probably be a prototype or an enum

**Comment by EmoGarbage404** [comment]
someones going to hell for that conflict

**Comment by Visne** [comment]
Direct connect button appears to get stuck when trying to connect to a non existing server. After the timeout it doesn't re-enable.

File: Content.Client/Actions/ActionsSystem.cs
Comment (DrSmugleaf) [reply Content.Client/Actions/ActionsSystem.cs]: Done in [`8b77ff2` (#7202)](https://github.com/space-wizards/space-station-14/pull/7202/commits/8b77ff2f831cc4fca8ce4747a39c0f2c9da58591)

**Comment by moonheart08** [comment]
all bugs fixed?
merg when

**Comment by DrSmugleaf** [comment]
> all bugs fixed? merg when

when i tell my gf that its done as a surprise tomorrow

**Comment by DrSmugleaf** [comment]
giphy

**Pull request merged**
